semantic_conventions/trace/database.yaml

```
@@ -202,7 +202,7 @@ groups:
             If applicable and not explicitly disabled via instrumentation configuration.
         brief: >
           The database statement being executed.
-        note: The value may be sanitized to exclude sensitive information.
+        note: The value may be sanitized to exclude sensitive information. It is recommended to sanitize the value by default in order to prevent security vulnerabilities.
```

I wonder if we shouldn't be more strict. Something like

> The value MUST be sanitized by default to avoid leaking sensitive information. Instrumentations may provide ways to turn off sanitation by the user.

---

+1
I think leaking PII makes the intent clearer.
"prevent security vulnerabilities" sounds as if it was more about preventing SQL injection and such which is unrelated to this.

---

It's not clear who this recommendation or even MUST is directed at.
Would this be the instrumentation's responsibility or should this be done by SDKs/processors/exporters/consumers?

---

I was thinking it should be part of the instrumentation.

what do you think of the following:
`note: The value should be sanitized by default to exclude sensitive information. Instrumentations may provide ways to turn off sanitization by the user.`

---

@arminru @joaopgrassi what do you think?

---

How about this?

```suggestion
        note: This attribute may contain private or sensitive information. It is recommended to sanitize it to exclude such information. Ideally, this is already done by default by the instrumentation collecting it.
```

Note that a statement like this is not of any normative nature, however, but it stresses the problem to make instrumentation authors aware of it.